### PR TITLE
Minor Turf Refactor

### PR DESCRIPTION
**Pull request by Fox-McCloud**
mostly from TG.

Turfs uhhhhh, aren't exactly going through the GC, in most cases.

I have no idea what kind of issues this is or is not creating, on line, but the only time turfs ever have their `Destroy` proc called is if they they are manually `qdel`'d, which is almost never how you interact with turfs.

This also refactors turfs to use `Initialize`--the sole exception is simulate turfs, for having air--otherwise, there will be mass runtimes; the air placement could be moved to `Initialize`, but the issue is that we don't have a good way to start tracking when a map has begun loading and finished---until we do, having `/turf/simulated` generate its air when it loads is the better deal.

### CODE REVIEW

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -70,16 +77,29 @@
 	log_startup_progress(" Smoothed atoms in [stop_watch(watch)]s.")
 	return 1
 
-/turf/Destroy()
-// Adds the adjacent turfs to the current atmos processing
+/turf/Destroy(force)
+	. = QDEL_HINT_IWILLGC
```
Comment (Fox-McCloud) [inline code/game/turfs/turf.dm:81]: You can't really delete turfs---one instance is invariably replaced with another

File: code/game/turfs/turf.dm [hunk 2]
```diff
@@ -45,7 +45,12 @@
 		queue_smooth(src)
 
 /turf/Initialize(mapload)
-	. = ..()
+	if(initialized)
+		stack_trace("Warning: [src]([type]) initialized multiple times!")
+	initialized = TRUE
```
Comment (AffectedArc07) [inline code/game/turfs/turf.dm:50]: imo this should be right before the return

File: code/game/turfs/turf.dm [hunk 3]
```diff
@@ -45,7 +45,12 @@
 		queue_smooth(src)
 
 /turf/Initialize(mapload)
-	. = ..()
+	if(initialized)
+		stack_trace("Warning: [src]([type]) initialized multiple times!")
```
Comment (AffectedArc07) [inline code/game/turfs/turf.dm:49]: `stack_trace()` does not halt execution, dont know if this is intended or not

File: code/game/turfs/simulated/floor.dm [hunk 4]
```diff
@@ -32,8 +32,8 @@ GLOBAL_LIST_INIT(icons_to_ignore_at_floor_init, list("damaged1","damaged2","dama
 	var/list/burnt_states = list("floorscorched1", "floorscorched2")
 	var/list/prying_tool_list = list(TOOL_CROWBAR) //What tool/s can we use to pry up the tile?
 
-/turf/simulated/floor/New()
-	..()
+/turf/simulated/floor/Initialize(mapload)
```
Comment (AffectedArc07) [inline code/game/turfs/simulated/floor.dm:35]: *oh lord*

File: code/game/turfs/turf.dm (hunk 3 above)
Comment (Fox-McCloud) [reply code/game/turfs/turf.dm:44]: Intended; we want want this logged; not halted.

atom initializations are the same.

**Pull request merged**